src/java.base/share/classes/java/lang/package-info.java

```diff
@@ -25,35 +25,42 @@
+ * Object}, which is the root of the class hierarchy, and {@link
  * Class}, instances of which represent classes at run time.
  *
  * <p>Frequently it is necessary to represent a value of primitive
- * type as if it were an object. The wrapper classes {@code Boolean},
- * {@code Character}, {@code Integer}, {@code Long}, {@code Float},
- * and {@code Double} serve this purpose.  An object of type {@code
- * Double}, for example, contains a field whose type is double,
+ * type as if it were an object.<a id="wrapperClasses"></a>
+ * The <dfn>{@index "wrapper classes"}</dfn> {@link Boolean},
+ * {@link Character}, {@link Short}, {@link Integer}, {@link Long}, {@link Float},
+ * and {@link Double} serve this purpose.  An object of type {@code
+ * Double}, for example, contains a field whose type is {@code double},
  * representing that value in such a way that a reference to it can be
- * stored in a variable of reference type.  These classes also provide
+ * stored in a variable of reference type.
+ * As discussed in <cite>The Java Language Specification</cite>, the
+ * wrapper classes are involved in boxing (JLS {@jls 5.1.7}) and
+ * unboxing (JLS {@jls 5.1.8}) conversions.
+ * These classes also provide
  * a number of methods for converting among primitive values, as well
- * as supporting such standard methods as equals and hashCode.  The
- * {@code Void} class is a non-instantiable class that holds a
- * reference to a {@code Class} object representing the type void.
+ * as supporting such standard methods as {@code equals} and {@code hashCode}.  The
+ * {@link Void} class is a non-instantiable class that holds a
+ * reference to a {@link Class} object representing the type {@code void}.
  *
- * <p>The class {@code Math} provides commonly used mathematical
- * functions such as sine, cosine, and square root. The classes {@code
- * String}, {@code StringBuffer}, and {@code StringBuilder} similarly
+ * <p>The class {@link Math} provides commonly used mathematical
+ * functions such as {@linkplain Math#sin(double) sine},
+ * {@linkplain Math#cos(double) cosine}, and
+ * {@linkplain Math#sqrt(double) square root}. The classes {@link
+ * String}, {@link StringBuffer}, and {@link StringBuilder} similarly
  * provide commonly used operations on character strings.
  *
- * <p>Classes {@code ClassLoader}, {@code Process}, {@code
- * ProcessBuilder}, {@code Runtime}, {@code SecurityManager}, and
- * {@code System} provide "system operations" that manage the dynamic
+ * <p>Classes {@link ClassLoader}, {@link Process}, {@link
+ * ProcessBuilder}, {@link Runtime}, {@link SecurityManager}, and
+ * {@link System} provide "system operations" that manage the dynamic
  * loading of classes, creation of external processes, host
  * environment inquiries such as the time of day, and enforcement of
  * security policies.
  *
- * <p>Class {@code Throwable} encompasses objects that may be thrown
+ * <p>Class {@link Throwable} encompasses objects that may be thrown
```

Comment:
We already have a link to `Class` in this doc comment. This one can stay `{@code Class}`, which might be easier to read.

Reply:
Isn't that the first reference to "Throwable" and the only link?

Reply:
Sorry, I sloppily positioned my previous comment. It's about `Class` not `Throwable`. `@link` to `Class` appears on L29-30 and L46.